### DISCUSSION TITLE
File hierarchy of devenv and application build wrapper on top of devenv

### DISCUSSION THREAD

**Discussion by tai271828** [solvcon/devenv#16] [General (discussions)]
Following the discussion of https://github.com/solvcon/devenv/issues/13 and https://github.com/solvcon/devenv/pull/15 , this thread will be used for discussion related to file hierarchy, especially the application build wrapper (will be) on top of `devenv`, e.g. `bin/build-application-solvcon-devenv.sh`.

In short, currently the features of `devenv` (the management tool) and the application build wrapper to build our target application e.g. `bin/build-application-solvcon-devenv.sh` are orthogonal. There are several points of view that we need to think about:

1. The features are orthogonal. Which folder they should share with each, and which folder should be separated.
1. If the folder names differ, what they should be?
1. The name of `build-application-solvcon-devenv.sh` looks nagging. Looking for a better name.
1. Is `applications/<application name>` a good naming? The other candidates are `solvcon`, `application-solvcon`, or `examples/solvcon`. See https://github.com/solvcon/devenv/issues/13#issuecomment-820741629 for more details.

**Comment by yungyuc** [comment]
> Following the discussion of https://github.com/solvcon/devenv/issues/13 and https://github.com/solvcon/devenv/pull/15 , this thread will be used for discussion related to file hierarchy, especially the application build wrapper (will be) on top of `devenv`, e.g. `applications/solvcon`.

Thanks for initiating this!  Could you help start the discussions by elaborating the challenges you described in https://github.com/solvcon/devenv/pull/15#issuecomment-821919532 ?

**Reply by tai271828** [reply]
Sounds good. I have updated the original description.

**Comment by yungyuc** [comment]
@j8xixo12 @chestercheng any comments you have?
